src/Temporalio.Extensions.Hosting/TemporalWorkerService.cs

```diff
@@ -19,6 +19,7 @@ public class TemporalWorkerService : BackgroundService
         private readonly TemporalClientConnectOptions? newClientOptions;
         private readonly ITemporalClient? existingClient;
         private readonly TemporalWorkerOptions workerOptions;
+        private readonly TemporalWorkerClientUpdater? workerClientUpdater;
```

Comment:
```suggestion
        private readonly ITemporalClient? existingClient;
        private readonly TemporalWorkerServiceOptions workerOptions;
```

Now that we are basically storing the three separate pieces anyways, let's just store the full service options as one field

Reply:
There's not really a good way I've found to handle the constructors that take in `TemporalWorkerOptions` directly (from some testing, cloning is not contravariant).

Reply:
Ah, I see now that we have multiple constructors for this `TemporalWorkerService` that accept partial forms. Ok, can ignore my comment and leave these as separate fields.